mqclient/broker_clients/rabbitmq.py

```diff
@@ -23,9 +23,56 @@
     Sub,
 )
 
+StrDict = Dict[str, Any]
+
 LOGGER = logging.getLogger("mqclient.rabbitmq")
 
 
+HUMAN_PATTERN = "[SCHEME://][USER[:PASS]@]HOST[:PORT][/VIRTUAL_HOST]"
+
+
+def _parse_url(url: str) -> Tuple[StrDict, Optional[str], Optional[str]]:
+    if "://" not in url:
+        url = "//" + url
+    result = urllib.parse.urlparse(url)
+
+    parts = dict(
+        scheme=result.scheme,
+        hostname=result.hostname,
+        port=result.port,
+        virtual_host=result.path.lstrip("/"),
+    )
+    # for putting into ConnectionParameters filter ""/None (will rely on defaults)
+    parts = {k: v for k, v in parts.items() if v}  # host=..., etc.
+
+    # check validity
+    if not parts or "hostname" not in parts:
+        raise RuntimeError(f"Invalid address: {url} (format: {HUMAN_PATTERN})")
+    parts["host"] = parts.pop("hostname")
```

Comment:
Why rename this here, instead of on line 41?

Reply:
`host` is what pika calls it

Reply:
oh, I see your point